hack/test-olm.sh

```diff
@@ -0,0 +1,392 @@
+
+#stop_registry() {
+#	if [[ "$REGISTRY_PID" != "" ]]; then
+#		kill ${REGISTRY_PID}
+#	fi
+#}
+#
+#start_registry() {
+#
+#	REGISTRY_OUT=reg.log
+#	docker run -p 5000:5000 registry:latest >${REGISTRY_OUT} 2>&1 &
+#	REGISTRY_PID=$?
+#
+#	trap "stop_registry" INT TERM HUP EXIT SIGINT
+#}
+
+
+start_registry() {
+	reg_name='local-registry'
+	reg_port='5000'
+	running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
+	if [ "${running}" != 'true' ]; then
+	  docker run \
+		-d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
+		registry:2
+	fi
+}
+
+setup_src() {
+	if [[ ! -f $LOCAL_OLM/repo_init ]]; then
+		git clone https://github.com/operator-framework/operator-lifecycle-manager.git $LOCAL_OLM
+		pushd $LOCAL_OLM
+		#git checkout origin/release-4.5 -b release-4.5
+		echo "" >repo_init
+		popd
+
+	fi
+
+	if [[ ! -f $LOCAL_OPREG/repo_init ]]; then
+		git clone https://github.com/operator-framework/operator-registry $LOCAL_OPREG
+		pushd $LOCAL_OPREG
+		make
+		echo "" >repo_init
+		popd
+	fi
+
+
+}
+
+setup_utils() {
+	mkdir bin || true
+
+	if [[ ! -x bin/kubectl ]]; then
+		curl -L https://storage.googleapis.com/kubernetes-release/release/v1.18.2/bin/linux/amd64/kubectl -o bin/kubectl
+		chmod +x bin/kubectl
+	fi
+
+#	if [[ ! -x bin/kind ]]; then
+#		curl -sLo bin/kind "$(curl -sL https://api.github.com/repos/kubernetes-sigs/kind/releases/latest | jq -r '[.assets[] | select(.name == "kind-linux-amd64")] | first | .browser_download_url')"
+#
+#		chmod +x bin/kind
+#	fi
+
+	if [[ ! bin/minikube ]]; then
+		curl -Lo ./bin/minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && chmod +x ./bin/minikube
+	fi
+
+	export PATH=$PWD/bin:$PWD/$LOCAL_OPREG/bin:$PATH
+}
+
+start_cluster() {
+	#./kind-with-registry.sh
+
+	minikube start  --insecure-registry="${LOCAL_REGISTRY}"
+	minikube ssh 'sudo bash -c "echo '$HOST_IP' localhost.localdomain >>/etc/hosts"'
+	pushd $LOCAL_OLM
+	#kind create cluster
+	#kind export kubeconfig
+	make run-local
+	kubectl get nodes
+	popd
+}
+
+
+stop_cluster() {
+	minikube stop
+	minikube delete
+}
+docker_tag() {
+	local from="$1"
+	local to="$2"
+
+	#IMG=$(docker images | grep -F $from' ' | awk '{ print $3 }')
+	IMG=$(docker images --no-trunc --quiet "$from")
+	docker tag $IMG $to
+	docker push $to
+}
+
+make_olm_op_bundle_new() {
+
+
+	# make the operator
+	make container-build
+
+	docker_tag $REGISTRY/$NMO $LOCAL_REGISTRY/$NMO
+
+	# make local copy of the bundle
+
+	rm -rf tmp-manifest || true
+	mkdir -p tmp-manifest/node-maintenance-operator/manifests
+	mkdir -p tmp-manifest/node-maintenance-operator/metadata
+
+
+	cp -rf manifests/node-maintenance-operator/v0.6.0/*  tmp-manifest/node-maintenance-operator/manifests
+
+	#cp manifests/node-maintenance-operator/node-maintenance-operator.package.yaml tmp-manifest/node-maintenance-operator/
+
+
+	cat >tmp-manifest/node-maintenance-operator/metadata/annotations.yaml <<EOF
```

Comment:
Probably we should have a template somewhere for this (like we have for manifests) and a makefile target that handles instantiating them.

Maybe copy some ideas from https://github.com/openshift-kni/performance-addon-operators/blob/master/Makefile since we're trying to follow their example

Reply:
I need to fix up the csv, so that it points to a image in the local registry instead of quay. Also need to build a registry that points to the bundle in the local registry.

Reply:
done, now all these instances are generated by a separate script.